lib/ansible/modules/web_infrastructure/ansible_tower/tower_receive.py

```diff
@@ -0,0 +1,174 @@
+      default: []
+    credential:
+      description:
+        - List of credential names to export
+      default: []
+    notification_template:
+      description:
+        - List of notification template names to export
+      default: []
+    inventory_script:
+      description:
+        - List of inventory script names to export
+      default: []
+    inventory:
+      description:
+        - List of inventory names to export
+      default: []
+    project:
+      description:
+        - List of project names to export
+      default: []
+    job_template:
+      description:
+        - List of job template names to export
+      default: []
+    workflow:
+      description:
+        - List of workflow names to export
+      default: []
+
+requirements:
+  - "ansible-tower-cli >= 3.3.0"
+
+notes:
+  - Specifying a name of "all" for any asset type will export all items of that asset type.
+
+extends_documentation_fragment: tower
+'''
+
+EXAMPLES = '''
+- name: Export all tower assets
+  tower_receive:
+    all: True
+    tower_config_file: "~/tower_cli.cfg"
+
+- name: Export all inventories
+  tower_receive:
+    inventory:
+      - all
+
+- name: Export a job template named "My Template" and all Credentials
+  tower_receive:
+    job_template:
+      - "My Template"
+    credential:
+      - all
+'''
+
+RETURN = '''
+assets:
+    description: The exported assets
+    returned: success
+    type: dict
+    sample: [ {}, {} ]
+'''
+
+from ansible.module_utils.ansible_tower import tower_argument_spec, tower_auth_config, HAS_TOWER_CLI
+from ansible.module_utils.basic import AnsibleModule
+
+try:
+    from tower_cli.cli.transfer.receive import Receiver
+    from tower_cli.cli.transfer.common import SEND_ORDER
+    from tower_cli.utils.exceptions import TowerCLIError
+
+    from tower_cli.conf import settings
+    TOWER_CLI_HAS_EXPORT = True
+except ImportError:
+    TOWER_CLI_HAS_EXPORT = False
+
+
+def main():
+    argument_spec = tower_argument_spec()
+    argument_spec.update(dict(
+        all=dict(type='bool', default=False),
+        credential=dict(type='list', default=[]),
+        credential_type=dict(type='list', default=[]),
+        inventory=dict(type='list', default=[]),
+        inventory_script=dict(type='list', default=[]),
+        job_template=dict(type='list', default=[]),
+        notification_template=dict(type='list', default=[]),
+        organization=dict(type='list', default=[]),
+        project=dict(type='list', default=[]),
+        team=dict(type='list', default=[]),
+        user=dict(type='list', default=[]),
+        workflow=dict(type='list', default=[]),
+    ))
+
+    module = AnsibleModule(argument_spec=argument_spec, supports_check_mode=False)
+
+    if not HAS_TOWER_CLI:
+        module.fail_json(msg='ansible-tower-cli required for this module')
+
+    if not TOWER_CLI_HAS_EXPORT:
+        module.fail_json(msg='ansible-tower-cli version does not support export')
+
+    all = module.params.get('all')
```

Comment:
`all` is a Python builtin keyword, let's pick a different variable name here

Reply:
Ha... I would have expected one of the linters to pick this up. I changed the python variable to export_all.